weave.tf

```diff
@@ -0,0 +1,151 @@
+resource "kubernetes_deployment" "weave" {
+  metadata {
+    name = local.weave_app_name
+    labels = {
+      app = local.weave_app_name
+    }
+  }
+
+  count = var.weave_enabled ? 1 : 0
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = local.weave_app_name
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = local.weave_app_name
+        }
+      }
+
+      spec {
+        priority_class_name = kubernetes_priority_class.priority.metadata[0].name
+
+        container {
+          name = local.weave_app_name
+          image             = "${var.wandb_image}:${var.wandb_version}"
+          image_pull_policy = "Always"
+
+          env {
+            name = "ONLY_SERVICE"
+            value = "weave"
+          }
+
+          env {
+            name = "WANDB_BASE_URL"
+            value = "${var.host}"
+          }
+
+          env {
+            name = "WEAVE_AUTH_GRAPHQL_URL"
+            value = "${var.host}/graphql"
+          }
+
+          env {
+            name = "DD_SERVICE"
+            value = "weave-python"
+          }
+
+          env {
+            name = "DD_ENV"
+            value = var.dd_env
+          }
+
+          env {
+            name = "DD_TRACE_AGENT_HOSTNAME"
+            value_from {
+              field_ref {
+                field_path = "status.hostIP"
+              }
+            }
+          }
+
+          env {
+            name = "DD_AGENT_HOST"
+            value_from {
+              field_ref {
+                field_path = "status.hostIP"
+              }
+            }
+          }
```

Comment:
I'm a little worried weave might try to talk to an agent even if `WEAVE_ENABLE_DATADOG` is set to false.  Ideally we would only specify these env vars if we know we have an agent running.

Reply:
good call. added a dynamic block.